que_.py

```diff
@@ -10,14 +10,14 @@ def enqueue(self, value):
     def dequeue(self):
             try:
                 return self.dll.shift()
-            except:
+            except ValueError:
                 raise ValueError("Dequeue does not work on an empty queue")
 
     def peek(self):
         try: 
             return self.dll.tail.prev.value
-        except:
-            raise ValueError("Peek does not work on an empty queue")
+        except AttributeError:
+            raise AttributeError("Peek does not work on an empty queue")
```

Comment:
An Attribute error means that a property you are trying to access isn't a part of the object. Say you have an object that looks like so:
```python
class A:
    def __init__(self):
        self.prop1 = "foo"
        self.prop2 = "bar"
    def method1(self):
        return "baz"
```
This object (when instantiated) has the attributes .prop1, prop2, and method1.   So if try to access something else like:
```python
a = A()
a.bungalow()
```
You get and error like 
```
AttributeError: ....
```

I agree that ValueError doesn't really fit either, but ValueError doesn't mean very specifically one thing.

If you'd like to look here https://docs.python.org/3/library/exceptions.html and see if something else make sense to you i'm open to suggestions.  One that I didn't know existed that might fit is UserWarning or RuntimeWarning.  

Also, if you want to make your own exception that really means the right thing you could always do something like this:
```
class MyBetterNamedError(Exception):
    pass
```

And then use it however you like :-]

Reply:
Trying to find one that fits there is a bit tricky, correct me if im wrong, but I think attribute error is appropriate, as the only time it is triggered is when there is nothing to peek at, which means trying to access a value from NoneType which does not have a value attribute.

Reply:
You are correct that the underlying issue is None.attribute doesn't work.  But the user requesting the next item from the queue is not doing anything wrong to warrant an attribute error.  The user should get an error that reflects what they have done wrong.

There being nothing to peek at doesn't have anything to do with an Attribute, it just happens to be the way you have(nt) implemented it in the dll.